crates/starknet-devnet/src/main.rs

```diff
@@ -198,20 +204,51 @@ async fn main() -> Result<(), anyhow::Error> {
 
     info!("Starknet Devnet listening on {}", address);
 
-    if starknet_config.dump_on == Some(DumpOn::Exit) {
-        server.with_graceful_shutdown(shutdown_signal(api.clone())).await?
-    } else {
-        server.await?
+    let empty_handle: task::JoinHandle<()> = task::spawn(async {});
```

Comment:
This looks rather hacky with the empty task. We can discuss.

Reply:
I decided to do it that way to avoid all 4 combinations.

Now we have an empty mutable task that can be an actual task with a ctrl c handle and we have server.with_graceful_shutdown with a second ctrl c handler and tokio::join later to handle ctrl c in both tasks

Reply:
empty task was removed in 6008680dc565e3860937183787088199ee1bef1f